src/xenia/cpu/compiler/passes/constant_propagation_pass.cc

```diff
@@ -510,7 +510,10 @@ bool ConstantPropagationPass::Run(HIRBuilder* builder) {
               mul->Mul(i->src2.value);
 
               Value* add = i->src3.value;
-              i->Replace(&OPCODE_ADD_info, 0);
+              if (mul->type == VEC128_TYPE)
+                i->Replace(&OPCODE_VECTOR_ADD_info, 0);
```

Comment:
Replace flags with `FLOAT32_TYPE`, i.e. `i->Replace(&OPCODE_VECTOR_ADD_info, FLOAT32_TYPE);`

Reply:
Also, surround the if statements with curly braces :)